base/libgit2/repository.jl

```diff
@@ -76,16 +76,25 @@ end
 """ Returns a found object """
 function revparse(repo::GitRepo, objname::AbstractString)
     obj_ptr_ptr = Ref{Ptr{Void}}(C_NULL)
-    err = ccall((:git_revparse_single, :libgit2), Cint,
-            (Ptr{Ptr{Void}}, Ptr{Void}, Cstring), obj_ptr_ptr, repo.ptr, objname)
-    err != 0 && return nothing
+    @check ccall((:git_revparse_single, :libgit2), Cint,
+           (Ptr{Ptr{Void}}, Ptr{Void}, Cstring), obj_ptr_ptr, repo.ptr, objname)
     return GitUnknownObject(repo, obj_ptr_ptr[])
 end
 
 """ Returns id of a found object """
 function revparseid(repo::GitRepo, objname::AbstractString)
-    obj = revparse(repo, objname)
-    obj === nothing && return GitHash()
+    local obj
+    try
+        obj = revparse(repo, objname)
```

Comment:
There are a few ways we could go here, and the best option depends on how the surrounding code is structured and which direction we generally want to go in with these binding API's. Are all of the possible not-found conditions errors, or should it be recoverable? Exceptions for control flow in the latter case should be avoided, we could do something with a nullable and/or also return the status code so the callers can decide what to do if things aren't found.